### ISSUE TITLE
Re: -Uusesfio doesn't work

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#3452]
Migrated from [rt.perl.org#5867](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=5867) (status was 'resolved')

Searchable as RT5867$

**Comment by p5pRT** [comment]
### From Richard.Foley@t-online.de

BTW\. I think it was bounced originally purely against the amazingly
effective 'no perl string' clause\.

:\-\)

>     \- Kurt
> 
> On Mon\, Feb 19\, 2001 at 03&#8203;:16&#8203;:56PM \-0000\, perlbugtron@&#8203;bugs\.perl\.org wrote&#8203;:
> >
> >     This email address is for reporting Perlbug bugs via \.
> >
> >
> >
> >     Please address your mail appropriately and include appropriate data
> >     as per the distributed documentation\.
> >
> >     Hint&#8203;: mail should include output from 'perl \-V'"
> >     \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> >     Hi\,
> >
> >     Pardon my brevity\.  Here's a passthrough of a bug from Debian&#8203;:
> >
> >     http&#8203;://bugs\.debian\.org/86207
> >
> >     \- Kurt
> >
> >
> >

\# Here's the bug report&#8203;:

Debian Bug report logs \- \#86207
perl is linked with libsfio when build on a
system with sfio installed

Package&#8203;: perl\-base; ;Reported by&#8203;: Roderich Schupp \<rsch@&#8203;ExperTeam\.de>;
Done&#8203;: Brendan O'Dea
\<bod@&#8203;debian\.org>; dated Fri\, 16 Feb 2001 19&#8203;:03&#8203;:46 UTC; Maintainer for
perl\-base is Brendan O'Dea
\<bod@&#8203;debian\.org>\. 

Report forwarded to debian\-bugs\-dist@&#8203;lists\.debian\.org\, Brendan O'Dea \<bod@&#8203;debian\.org>&#8203;:
Bug\#86207; Package perl\-base\. Full text available\. 

Acknowledgement sent to Roderich Schupp \<rsch@&#8203;ExperTeam\.de>&#8203;:
New Bug report received and forwarded\. Copy sent to Brendan O'Dea
\<bod@&#8203;debian\.org>\. Full text
available\. 


Message received at submit@&#8203;bugs\.debian\.org&#8203;:


Received&#8203;: \(at submit\) by bugs\.debian\.org; 16 Feb 2001 18&#8203;:42&#8203;:14 \+0000
From rsch@&#8203;experteam\.de Fri Feb 16 12&#8203;:42&#8203;:14 2001
Return\-path&#8203;: \<rsch@&#8203;experteam\.de>
Received&#8203;: from www1\.experteam\.de \[&#8203;::ffff&#8203;:195\.138\.53\.252\] 
&nbsp;        by master\.debian\.org with esmtp \(Exim 3\.12 1 \(Debian\)\)
&nbsp;        id 14Tppd\-000879\-00; Fri\, 16 Feb 2001 12&#8203;:42&#8203;:13 \-0600
Received&#8203;: from m\-kombi\.ExperTeam\.de \(m\-kombi\.experteam\.de \[192\.168\.5\.1\]\)
&nbsp;        by www1\.ExperTeam\.de \(8\.9\.1/8\.9\.3\) with SMTP id TAA15148
&nbsp;        for \<submit@&#8203;bugs\.debian\.org>; Fri\, 16 Feb 2001 19&#8203;:42&#8203;:10 \+0100
Received&#8203;: from ExperTeam\.de \(\[192\.168\.5\.52\]\) by m\-kombi\.ExperTeam\.de
\(Lotus SMTP MTA v4\.6\.2  \(693\.3 8\-11\-1998\)\) with SMTP id
412569F5\.00680F67; Fri\, 16 Feb 2001 19&#8203;:56&#8203;:36 \+0100
Sender&#8203;: rsch@&#8203;ExperTeam\.de
Message\-ID&#8203;: \<3A8D6DB0\.640E653F@&#8203;ExperTeam\.de>
Date&#8203;: Fri\, 16 Feb 2001 19&#8203;:13&#8203;:04 \+0100
From&#8203;: Roderich Schupp \<rsch@&#8203;ExperTeam\.de>
Organization&#8203;: ExperTeam AG
X\-Mailer&#8203;: Mozilla 4\.76 \[en\] \(X11; U; Linux 2\.4\.2\-pre3\+nfsdops i586\)
X\-Accept\-Language&#8203;: en
MIME\-Version&#8203;: 1\.0
To&#8203;: submit@&#8203;bugs\.debian\.org
Subject&#8203;: perl is linked with libsfio when build on a system with sfio installed
Content\-Type&#8203;: text/plain; charset=us\-ascii
Content\-Transfer\-Encoding&#8203;: 7bit
Delivered\-To&#8203;: submit@&#8203;bugs\.debian\.org

Package&#8203;: perl\-base
Version&#8203;: 5\.6\.0\-17
Severity&#8203;: normal

Subject says it all\. \-Uusesfio is not enough to prevent detection
an subsequent link of libsfio\. Fix&#8203;:

<details><summary>Inline Patch</summary>

```diff
--- ORIG/debian/config.over     Fri Feb 16 08:59:49 2001
+++ ./debian/config.over        Fri Feb 16 09:00:14 2001
@@ -21,12 +21,12 @@
 
 ccdlflags="$tmp"
 
-# remove -l*dbm and -ldb libs
+# remove -l*dbm, -ldb and -lsfio libs
 tmp=
 for t in $libs
 do
     case $t in
-       -l*dbm|-ldb) ;;
+       -l*dbm|-ldb|-lsfio) ;;
        *) tmp="$tmp${tmp:+ }$t"
     esac
 done




```
</details>
Cheers, Roderich




Reply sent to Brendan O'Dea \<bod@&#8203;debian\.org>&#8203;:
You have taken responsibility\. Full text available\. 

Notification sent to Roderich Schupp \<rsch@&#8203;ExperTeam\.de>&#8203;:
Bug acknowledged by developer\. Full text available\. 


Message received at 86207\-close@&#8203;bugs\.debian\.org&#8203;:


Received&#8203;: \(at 86207\-close\) by bugs\.debian\.org; 18 Feb 2001 20&#8203;:11&#8203;:34 \+0000
From troup@&#8203;auric\.debian\.org Sun Feb 18 14&#8203;:11&#8203;:34 2001
Return\-path&#8203;: \<troup@&#8203;auric\.debian\.org>
Received&#8203;: from auric\.debian\.org \[&#8203;::ffff&#8203;:206\.246\.226\.45\] 
&nbsp;        by master\.debian\.org with esmtp \(Exim 3\.12 1 \(Debian\)\)
&nbsp;        id 14UaBA\-0003ss\-00; Sun\, 18 Feb 2001 14&#8203;:11&#8203;:33 \-0600
Received&#8203;: from troup by auric\.debian\.org with local \(Exim 3\.12 1 \(Debian\)\)
&nbsp;        id 14Ua7A\-0003Iw\-00; Sun\, 18 Feb 2001 15&#8203;:07&#8203;:24 \-0500
From&#8203;: Brendan O'Dea \<bod@&#8203;debian\.org>
To&#8203;: 86207\-close@&#8203;bugs\.debian\.org
Subject&#8203;: Bug\#86207&#8203;: fixed in perl 5\.6\.0\-18
Message\-Id&#8203;: \<E14Ua7A\-0003Iw\-00@&#8203;auric\.debian\.org>
Sender&#8203;: James Troup \<troup@&#8203;auric\.debian\.org>
Date&#8203;: Sun\, 18 Feb 2001 15&#8203;:07&#8203;:24 \-0500
Delivered\-To&#8203;: 86207\-close@&#8203;bugs\.debian\.org

We believe that the bug you reported is fixed in the latest version of
perl\, which has been installed in the Debian FTP archive&#8203;:

perl\-doc\_5\.6\.0\-18\_all\.deb
&nbsp;  to pool/main/p/perl/perl\-doc\_5\.6\.0\-18\_all\.deb
perl\-suid\_5\.6\.0\-18\_i386\.deb
&nbsp;  to pool/main/p/perl/perl\-suid\_5\.6\.0\-18\_i386\.deb
libperl5\.6\_5\.6\.0\-18\_i386\.deb
&nbsp;  to pool/main/p/perl/libperl5\.6\_5\.6\.0\-18\_i386\.deb
perl\_5\.6\.0\-18\.dsc
&nbsp;  to pool/main/p/perl/perl\_5\.6\.0\-18\.dsc
libperl\-dev\_5\.6\.0\-18\_i386\.deb
&nbsp;  to pool/main/p/perl/libperl\-dev\_5\.6\.0\-18\_i386\.deb
perl\_5\.6\.0\-18\.diff\.gz
&nbsp;  to pool/main/p/perl/perl\_5\.6\.0\-18\.diff\.gz
libcgi\-fast\-perl\_5\.6\.0\-18\_all\.deb
&nbsp;  to pool/main/p/perl/libcgi\-fast\-perl\_5\.6\.0\-18\_all\.deb
perl\_5\.6\.0\-18\_i386\.deb
&nbsp;  to pool/main/p/perl/perl\_5\.6\.0\-18\_i386\.deb
perl\-debug\_5\.6\.0\-18\_i386\.deb
&nbsp;  to pool/main/p/perl/perl\-debug\_5\.6\.0\-18\_i386\.deb
perl\-modules\_5\.6\.0\-18\_all\.deb
&nbsp;  to pool/main/p/perl/perl\-modules\_5\.6\.0\-18\_all\.deb
perl\-base\_5\.6\.0\-18\_i386\.deb
&nbsp;  to pool/main/p/perl/perl\-base\_5\.6\.0\-18\_i386\.deb
A summary of the changes between this version and the previous one is
attached\.

Thank you for reporting the bug\, which will now be closed\.  If you
have further comments please address them to 86207@&#8203;bugs\.debian\.org\,
and the maintainer will reopen the bug report if appropriate\.

Debian distribution maintenance software
pp\.
Brendan O'Dea \<bod@&#8203;debian\.org> \(supplier of updated perl package\)

\(This message was generated automatically at their request; if you
believe that there is a problem with it please contact the archive
administrators by mailing ftpmaster@&#8203;debian\.org\)


\-\-\-\-\-BEGIN PGP SIGNED MESSAGE\-\-\-\-\-
Hash&#8203;: SHA1

Format&#8203;: 1\.7
Date&#8203;: Sun\, 18 Feb 2001 06&#8203;:13&#8203;:29 \+1100
Source&#8203;: perl
Binary&#8203;: perl\-suid perl\-modules perl perl\-debug perl\-base libperl5\.6
perl\-doc libperl\-dev libcgi\-fast\-perl
Architecture&#8203;: source i386 all
Version&#8203;: 5\.6\.0\-18
Distribution&#8203;: unstable
Urgency&#8203;: low
Maintainer&#8203;: Brendan O'Dea \<bod@&#8203;debian\.org>
Changed\-By&#8203;: Brendan O'Dea \<bod@&#8203;debian\.org>
Description&#8203;: 
 libcgi\-fast\-perl \- CGI&#8203;::Fast Perl module\.
 libperl\-dev \- Perl library&#8203;: development files\.
 libperl5\.6 \- Shared Perl library\.
 perl       \- Larry Wall's Practical Extracting and Report Language\.
 perl\-base  \- The Pathologically Eclectic Rubbish Lister\.
 perl\-debug \- Debug\-enabled Perl interpreter\.
 perl\-doc   \- Perl documentation\.
 perl\-modules \- Core Perl modules\.
 perl\-suid  \- Runs setuid Perl scripts\.
Closes&#8203;: 86059 86122 86154 86204 86207 86364
Changes&#8203;: 
 perl \(5\.6\.0\-18\) unstable; urgency=low
 \.
&nbsp;   \* Updated to debhelper v3\.
&nbsp;   \* Include changelogs in libcgi\-fast\-perl package\.  \(Closes&#8203;: \#86364\)\.
&nbsp;   \* Remove empty B&#8203;::Stash and warnings&#8203;::register man pages\.
&nbsp;     \(Closes&#8203;: \#86204\)\.
 \.
&nbsp;   \* Fix Math&#8203;::Complex manual page \(11\_fix\_math\_complex\_pod\)\.
&nbsp;   \* Remove \-lsfio library\.  \(Closes&#8203;: \#86207\)\.
&nbsp;   \* Move AUTHORS to perl\-base\, as copyright references it\.  Thanks to
&nbsp;     Dirk Eddelbuettel \<edd@&#8203;debian\.org>\.
 \.
&nbsp;   \* Move perldiag\.pod to the perl\-modules package so that the
&nbsp;     diagnostics module works without perl\-doc being installed\.
&nbsp;     \(Closes&#8203;: \#86154\)\.
 \.
&nbsp;   \* Add /usr/lib/perl5/5\.6 to @&#8203;INC to handle arch\-indep module packages
&nbsp;     which didn't set privlib to /usr/lib/perl5\.  \(Closes&#8203;: \#86122\)\.
 \.
&nbsp;   \* Add \-mieee for alpha build\.  \(Closes&#8203;: \#86059\)\.
Files&#8203;: 
 0758ad650ada527e25b91489d1975b55 726 interpreters standard perl\_5\.6\.0\-18\.dsc
 1278d107c338aee9851b6d997fa39e0c 45135 interpreters standard perl\_5\.6\.0\-18\.diff\.gz
 eb3242893b55659f7fa0ab53e08c48ca 299208 interpreters optional libcgi\-fast\-perl\_5\.6\.0\-18\_all\.deb
 e6be18183808ab722a2b018c16e24b5e 3642660 doc optional perl\-doc\_5\.6\.0\-18\_all\.deb
 b560802f7895e632c40eab7ed31c6778 1201482 interpreters standard perl\-modules\_5\.6\.0\-18\_all\.deb
 6bfcd5c71cd0161e1cac8b6a76a11b43 833424 base required perl\-base\_5\.6\.0\-18\_i386\.deb
 7bffe5ffbe300fbb73aa5068976aba41 351384 interpreters optional perl\-debug\_5\.6\.0\-18\_i386\.deb
 2aa94aa312a328e15d41eaa00b230f19 28162 interpreters optional perl\-suid\_5\.6\.0\-18\_i386\.deb
 bdd5339472d7e3d56c62c10da243103f 335206 libs optional libperl5\.6\_5\.6\.0\-18\_i386\.deb
 a9a676bbf670c65bc9b17d30ed95c65e 411212 devel optional libperl\-dev\_5\.6\.0\-18\_i386\.deb
 769760dbd0453a9e5f78bb5e4292c862 763466 interpreters important perl\_5\.6\.0\-18\_i386\.deb

\-\-\-\-\-BEGIN PGP SIGNATURE\-\-\-\-\-
Version&#8203;: GnuPG v1\.0\.4 \(GNU/Linux\)
Comment&#8203;: For info see http&#8203;://www\.gnupg\.org

iD8DBQE6jxw5e7iP96nkr/IRAudQAKCIKqGnPXfegWpV64MBbtkod6CyoACffa0L
585ec\+HopwAJluGrLJ5JXTo=
=y4Kw
\-\-\-\-\-END PGP SIGNATURE\-\-\-\-\-

**Comment by p5pRT** [comment]
### From @jhi

\<sigh> That message mentions also \*other\* bug fixes made to be Debian
Perl packages\, fixes that have not made their way to the standard
distribution\(s\)\.  Anyone feeling like taking a stroll through the
Debian bug databases?
